### ISSUE TITLE
Extract macro meta-data

### ISSUE THREAD

**Issue by vi-h** [enonic/lib-guillotine#71]
When using a macro inside HtmlArea, and getting it via GraphQL, the processHtml returns the macros inside a comment.
**Ex:**
_Result when using processHtml() from lib-portal:_
```
<p>Articleblock which should refer to another site with an anchor: <a href="/admin/portal/preview/draft/kss/min-artikkel#addedanchor" target="_self" title="Tooltip">Text for the anchor</a></p>

<p>And a link which works: <a href="\" title="\&quot;Tooltip\&quot;">Testpage link</a></p>
```

_Result when using processHtml() to get it in GraphQL_
```
"<p>Articleblock which should refer to another site with an anchor: <!--#MACRO _name=\"link-with-anchor\" anchorTag=\"addedanchor\" tooltip=\"Tooltip\" linktext=\"Text for the anchor\" content=\"5330118b-16ee-484d-82a0-77d6a2e896b3\" _document=\"__macroDocument1\" _body=\"\"--></p>\n\n<p>And a link which works: <a href=\"https://localhost:8080/admin/portal/preview/draft/mysite/test/testside\" title=\"Tooltip\">Testpage link</a></p>\n"
```

**Comment by alansemenov** [comment]
lib-guillotine's `processHtml` is directly calling portal-lib's `processHtml`, so there's no way you'd get different responses from those two. But if you are calling portal.processHtml _inside a template_, like so
```HTML
<div data-th-remove="tag" data-th-utext="${portal.processHtml({'_value=' + htmlareavalue})}"></div>
````
then yes, macro will be processed, but only because in this case it's going through the _rendering engine_.

If you call processHtml in your Javascript controller then it will give you the same unprocessed macro you specified in your bug description.

Here's the test I've just made:
```Javascript
var portal = require('/lib/xp/portal');

exports.get = function() {

    var content = portal.getContent();

    log.info('Unprocessed: ' + content.data.htmlarea_text);
    log.info('processHtml: ' + portal.processHtml({value: content.data.htmlarea_text}));

    return {body: ''};
};
``` 

![image](https://user-images.githubusercontent.com/11612490/100244460-3b540180-2f37-11eb-97a6-55e5efedbca4.png)

Long story short, in order to solve this we'll have to move processing of macros out of the rendering engine.

**Comment by sigdestad** [comment]
Alan, I have already discussed this in detail with @anatol-sialitski. Running macro's will be a separate parameter from processHtml. Also, there are considerations to rendering in headless context vs inline pages
